go-pkg/ds/permission.go

```diff
@@ -22,6 +28,62 @@ func (u *User) addKey(blob *protods.DataBlob, newDataOwner *User) *protods.DataB
 // 1. Generate an AES-key
 // 2. Add an entry to "Secrets"
 // 3. Encrypt data with AES key
-func (u *User) createDataBlob(data []byte) *protods.DataBlob {
-	panic("not implemented")
+func (u *User) createDataBlob(data []byte) (*protods.DataBlob, error) {
+	hiddenDataKey, AESKey, err := u.generateHiddenKey()
```

Comment:
The AesKey should be an input of to that function instead. The reason is that the `AesKey` could be re-used to generate multiple HiddenDataKeys, one for different users.